amazonka-test-1.4.4 build failure with aeson-1.0.2.1

As seen on the stackage build server:

```
[5 of 6] Compiling Test.AWS.Fixture ( src/Test/AWS/Fixture.hs, dist/build/Test/AWS/Fixture.o )

src/Test/AWS/Fixture.hs:126:14: error:
    • Overlapping instances for FromJSON
                                  (HashMap HeaderName ByteString)
        arising from a use of ‘.:?’
      Matching instances:
        instance (FromJSON v, FromJSONKey k, Eq k, Hashable k) =>
                 FromJSON (HashMap k v)
          -- Defined in ‘aeson-1.0.2.1:Data.Aeson.Types.FromJSON’
        ...plus one instance involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the first argument of ‘(.!=)’, namely ‘o .:? "headers"’
      In the first argument of ‘(<&>)’, namely
        ‘o .:? "headers" .!= mempty’
      In the second argument of ‘(<*>)’, namely
        ‘(o .:? "headers" .!= mempty <&> Map.toList)’
```

By passing the suggested ghc flag, we get the full message:

```bash
$ stack test --ghc-options -fprint-potential-instances
```

```
[5 of 6] Compiling Test.AWS.Fixture ( src/Test/AWS/Fixture.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.0.0/build/Test/AWS/Fixture.o )

/Users/danburton/scratch/amazonka-test-1.4.4/src/Test/AWS/Fixture.hs:126:14: error:
    • Overlapping instances for FromJSON
                                  (HashMap HeaderName ByteString)
        arising from a use of ‘.:?’
      Matching instances:
        instance (FromJSON v, FromJSONKey k, Eq k, Hashable k) =>
                 FromJSON (HashMap k v)
          -- Defined in ‘aeson-1.0.2.1:Data.Aeson.Types.FromJSON’
        ...plus one instance involving out-of-scope types
          instance FromJSON (HashMap HeaderName ByteString)
            -- Defined in ‘Test.AWS.Orphans’
    • In the first argument of ‘(.!=)’, namely ‘o .:? "headers"’
      In the first argument of ‘(<&>)’, namely
        ‘o .:? "headers" .!= mempty’
      In the second argument of ‘(<*>)’, namely
        ‘(o .:? "headers" .!= mempty <&> Map.toList)’
```

Therefore, the following instances overlap:

| module | instance |
|---|---|
| `Test.AWS.Orphans` | `HashMap HeaderName ByteString` |
| `aeson-1.0.2.1:Data.Aeson.Types.FromJSON` | `HashMap k v`  |

I suggest removing the orphan, or if you want backwards compatibilty, use CPP to conditially remove it based on the version of aeson. Alternatively, use a newtype to preserve your custom instance in a way that doesn't conflict with the instance provided by aeson.

--- a/core/src/Network/AWS/Data/Text.hs
+++ b/core/src/Network/AWS/Data/Text.hs
@@ -76,6 +76,9 @@ instance FromText Text where
 instance FromText ByteString where
     parser = Text.encodeUtf8 <$> A.takeText
 
+instance (CI.FoldCase a, FromText a) => FromText (CI a) where
+    parser = CI.mk <$> parser
+
 instance FromText Char where
     parser = A.anyChar <* A.endOfInput
 

--- a/stack-8.0.1.yaml
+++ b/stack-8.0.1.yaml
@@ -1,5 +1,5 @@
 compiler:   ghc-8.0.1
-resolver:   nightly-2016-08-11
+resolver:   nightly-2016-10-30
 flags:      {}
 extra-deps:
   - http-conduit-2.2.0

--- a/test/src/Test/AWS/Fixture.hs
+++ b/test/src/Test/AWS/Fixture.hs
@@ -32,6 +32,7 @@ import           Data.Proxy
 import           Data.Time
 import qualified Data.Yaml                    as YAML
 import           Network.AWS.Data.ByteString
+import qualified Network.AWS.Data.Map         as Map
 import           Network.AWS.Lens             (trying, (<&>))
 import           Network.AWS.Prelude
 import           Network.HTTP.Client.Internal hiding (Proxy, Request, Response)
@@ -123,7 +124,7 @@ instance FromJSON Req where
         <$> o .: "method"
         <*> (o .:? "path"    .!= "/")
         <*> (o .:? "query"   .!= mempty)
-        <*> (o .:? "headers" .!= mempty <&> Map.toList)
+        <*> (o .:? "headers" .!= mempty <&> Map.toList . Map.toMap)
         <*> (o .:? "body"    .!= mempty)
 
 sortKeys :: Ord a => [(a, b)] -> [(a, b)]

--- a/test/src/Test/AWS/Orphans.hs
+++ b/test/src/Test/AWS/Orphans.hs
@@ -14,28 +14,8 @@
 --
 module Test.AWS.Orphans where
 
-import           Data.Aeson
-import qualified Data.CaseInsensitive as CI
-import qualified Data.HashMap.Strict  as Map
-import qualified Data.Text.Encoding   as Text
-import           Data.Traversable     (traverse)
-import           Network.AWS.Prelude
+import Data.Aeson
+import Network.AWS.Prelude
 
 instance FromJSON ByteString where
     parseJSON = withText "bytestring" (either fail pure . fromText)
-
-instance FromJSON (HashMap ByteString ByteString) where
-    parseJSON = withObject "hashmap" $
-          fmap Map.fromList
-        . traverse go
-        . Map.toList
-      where
-        go (k, v) = (Text.encodeUtf8 k,) <$> parseJSON v
-
-instance FromJSON (HashMap HeaderName ByteString) where
-    parseJSON = withObject "headers" $
-          fmap Map.fromList
-        . traverse go
-        . Map.toList
-      where
-        go (k, v) = (CI.mk (Text.encodeUtf8 k),) <$> parseJSON v

